drop-transfer/src/service.rs

```diff
@@ -36,14 +38,28 @@ pub(super) struct State {
     pub fdresolv: Option<Arc<crate::file::FdResolver>>,
 }
 
+pub type PeerStates = HashMap<IpAddr, PeerState>;
+
+#[derive(Deserialize, Debug)]
+pub enum PeerState {
+    #[serde(rename = "online")]
+    Online,
+    #[serde(rename = "offline")]
+    Offline,
+}
+
+type TaskRegistry =
+    Arc<tokio::sync::Mutex<HashMap<(IpAddr, uuid::Uuid), tokio::task::JoinHandle<()>>>>;
```

Comment:
I don't think you need `IpAddr` where. It's unambiguously defined by the `Uuid` since transfers have only one peer (IP)

Reply:
Moreover, you never call `clone()` on it so `Arc` is not necessary here and adds unnecessary indirection

Reply:
Going deeper you don't need `Mutex`, since `Service` is being passed as a mutable reference

Reply:
You're right about the UUID and peer. This simplified things

Reply:
Mutex point is also truth, I simplified it